internal/service/news/news.go

```diff
@@ -0,0 +1,123 @@
+package news
+
+import (
+	gocontext "context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/newUser1337/task-news/internal/config"
+	"github.com/newUser1337/task-news/internal/context"
+	"github.com/newUser1337/task-news/internal/entity"
+	"github.com/newUser1337/task-news/internal/httpclient"
+	"github.com/newUser1337/task-news/internal/repository"
+)
+
+const articlesListEndpoint string = "https://www.htafc.com/api/incrowd/getnewlistinformation?count=50"
+const articleEndpoint string = "https://www.htafc.com/api/incrowd/getnewsarticleinformation?id=%d"
+
+type articleRepository interface {
+	SaveArticleList(gocontext.Context, []*entity.NewsletterNewsItemDB) ([]int64, error)
+	SaveArticle(gocontext.Context, *entity.NewsArticleDB) (int64, error)
+}
+
+type NewsServer struct {
+	appCtx  *context.AppCtx
+	article articleRepository
+}
+
+func StartNewNewsServer(appCtx *context.AppCtx, cfg *config.Config, rep *repository.Repository) {
+	newsServer := &NewsServer{
+		appCtx:  appCtx,
+		article: rep.Article,
+	}
+
+	go newsServer.maintain(cfg)
+}
+
+func (n *NewsServer) maintain(cfg *config.Config) {
+	n.appCtx.WgAdd(1)
+	defer n.appCtx.WgDone()
+
+	refreshTime := cfg.Core.RefreshInterval
+	ticker := time.NewTicker(refreshTime)
+	defer ticker.Stop()
+	if err := n.getArticles(); err != nil {
+		err = fmt.Errorf("news-srv: failed to get articels %s", err)
+		log.Println(err)
+	}
+
+	for range ticker.C {
+		select {
+		case <-n.appCtx.GetGoContext().Done():
+			return
+		default:
+		}
+
+		if err := n.getArticles(); err != nil {
+			err = fmt.Errorf("news-srv: failed to get articels %s", err)
+			log.Println(err)
+		}
+	}
+}
+
+func (n *NewsServer) getArticles() error {
+	httpClient := httpclient.NewClient(httpclient.WithTimeout(time.Second * 5))
+
+	articlesExt := &entity.ArticleListExternal{}
+	if err := httpClient.FetchData(articlesListEndpoint, articlesExt); err != nil {
```

Comment:
hmm, why do you not return `articlesExt` from `FetchData`? 🤔

Reply:
Cuase i use FetchData for &entity.ArticleListExternal{} and &entity.ArticleExternal{} types